### PR TITLE
Fix CI issues

### PR DESCRIPTION
**Pull request by AshishPvjs**
- Remove support for Python 3.7 and add support for Python 3.11
- Update CI test environment to drop Python 3.7 and Python 3.11
- Fix typos in docstrings for fairness metrics get_scores method.
- Update S3 link in evalrs lfm_dataset_path

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (skadio) [inline .github/workflows/ci.yml:18]: quick q -- does jurity work with 3.12?

Comment (AshishPvjs) [reply .github/workflows/ci.yml:18]: I need to run some tests before confirming this.

File: CHANGELOG.txt [hunk 2]
```diff
@@ -2,6 +2,15 @@
 CHANGELOG
 =========
 
+-------------------------------------------------------------------------------
+Sep 06, 2024 2.1.0
+-------------------------------------------------------------------------------
+Major
+- Remove support for Python 3.7 and add support for Python 3.11
```
Comment (skadio) [inline CHANGELOG.txt:9]: don't you need to update README now?

Comment (skadio) [reply CHANGELOG.txt:9]: and possibly docs?

File: evalrs/evaluation/utils.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 from datetime import datetime
 
 
-LFM_DATASET_PATH="https://cikm-evalrs-dataset.s3.us-west-2.amazonaws.com/evalrs_dataset.zip"
+LFM_DATASET_PATH="https://evarl-2022-public-dataset.s3.us-east-1.amazonaws.com/evalrs_dataset.zip"
```
Comment (skadio) [inline evalrs/evaluation/utils.py:20]: this is good, let's make this update. And at a second look, can we follow-up with another PR right after to nuke the evalrs folder? The users of this library does not really need this, right? Why would a general jurity users inherits all this when they install the library (back then we didn't have a place to store for the competition, hence it's addition here but now that's over and not needed). 

My recommendation is, proceed with this PR as is to fix the dangling dataset path. Then follow-up with an immediate PR to remove the evalrs folder. We can move that folder under huggingface/skadio

File: .github/workflows/ci.yml (hunk 1 above)
Comment (skadio) [reply .github/workflows/ci.yml:18]: while at it, jumping full onboard would be nice to avoid going one-by-one

**Review by skadio — APPROVED** [review]
LGTM -- a few clean-up comments only. I think we can decommission the evalrs folder fully once we merge the fix here

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AshishPvjs) [reply .github/workflows/ci.yml]: Jurity now supports python 3.12

File: CHANGELOG.txt (hunk 2 above)
Comment (AshishPvjs) [reply CHANGELOG.txt]: Updated readme, docs & quick start

**Pull request merged**